docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,603 @@
+---
+lang: en
+title: 'Authentication'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Loopback-component-authentication.html
+---
+
+## Overview
+
+Security is of paramount importance when developing a web or mobile application
+and usually consists of two distinct pieces :
+
+- Authentication
+- Authorization
+
+`Authentication` is a process of verifying someone's identity before a protected
+endpoint is accessed.
+
+`Authorization` is a process of verifying the actions that an authenticated user
+is allowed to perform on a protected endpoint.
+
+{% include note.html content="
+For a description of an `Authorization` process, please see <a href='https://loopback.io/doc/en/lb4/Loopback-component-authorization.html'>Authorization</a>.
+" %}
+
+This document describes the details of the LoopBack 4 `Authentication`
+component.
+
+![authentication_overview](./imgs/authentication_overview.png) Basically, to
```

Comment:
Thanks for adding the diagram. I was thinking of a more higher-level diagram to help users to understand the big picture first. :)
I was thinking something along the line of the diagram below (it's probably not reflecting what's actually happening, but just an illustration). Does it make sense? 
<img width="303" alt="Screen Shot 2019-06-07 at 5 36 07 PM" src="https://user-images.githubusercontent.com/25489897/59134977-43873e00-894b-11e9-9868-2c186e6337c9.png">

Reply:
Perhaps we should have a `whiteboard session` and see how we can move between highlevel and lowlevel diagrams and when...

Reply:
It's just a suggestion.  Since I don't have much background on the detailed design, so a higher level diagram helps me better to understand.  Maybe it's just personal preference.

Reply:
@dhmlau 

The action provider isn't a composite part of the strategy provider, and the metadata provider can be used from anywhere (actually all of them are set up like this), so it is more accurate to have them separate from each other than contained in each other.

Reply:
@dhmlau @jannyHou how about this for a high level?
![image](https://user-images.githubusercontent.com/6864736/59280648-eaadf300-8c33-11e9-8747-026e9ae6e8bd.png)

Reply:
1. wondering if I should show custom authentication strategies at all in this diagram and just leave it as `extensions`
2. wondering if I should specify the parameters of the decorator: strategyName:string, options?:object .

Reply:
actually this one is better

![image](https://user-images.githubusercontent.com/6864736/59283703-f8b24280-8c38-11e9-944f-9e235c0e9802.png)